docs/gettinginvolved/contributing.rst

```diff
@@ -79,3 +79,100 @@ Similarly to `maintenance-branches`_, bug fixes and improvements may need to be
+
+Start with some nodes which are configured correctly for Flocker.
+A simple way to do this is to run the :ref:`acceptance test runner <acceptance-testing>` with the ``--keep`` option.
+
+Log in to each node in the cluster, forwarding the authentication agent connection:
+
+.. prompt:: bash alice@mercury$
+
+   ssh -A root@${NODE_IP}
+
+At this point it is possible to install packages built by `Buildbot`_.
+This would be a more complete testing of code on a node.
+However, it takes some time for packages to be built, and for development purposes the trade-off of a fast development cycle may be worthwhile.
+Those trade-offs include the ability to test new or changed Python dependencies.
+The following instructions replace just some of the code used by Flocker, but enough that can be useful.
+
+On each node, run the following commands.
+There are tools which allow you to run commands in multiple consoles simultaneously,
+such as ``iTerm2`` for Mac OS X:
+
+#. Install ``git``:
+
+   .. prompt:: bash [root@node1]$
+
+      sudo yum install -y git
+
+#. Clone Flocker somewhere to use later:
+
+   .. prompt:: bash [root@node1]$
+
+      mkdir /flocker-source
+      cd /flocker-source
+      git clone git@github.com:ClusterHQ/flocker.git
+
+#. Change the Flocker code in the checkout to what needs to be tested:
+
+   .. prompt:: bash [root@node1]$
+
+      cd /flocker-source/flocker
+      git checkout BRANCH-NAME
+
+#. Make a backup of the code and unit files which will be replaced:
+
+   .. prompt:: bash [root@node1]$
+
+      mkdir /backup
+      cp -r /opt/flocker/lib/python2.7/site-packages/flocker/ /backup
+      cp -r /etc/systemd/system/multi-user.target.wants/ /backup
+
+#. Replace the node services with the new code:
+
+   .. prompt:: bash [root@node1]$
+
+      # Move Python code from the Git clone to where they are used
+      rm -rf /opt/flocker/lib/python2.7/site-packages/flocker/
+      cp -r /flocker-source/flocker/flocker/ /opt/flocker/lib/python2.7/site-packages/
+
+      # Stop systemd units before they are changed
+      cd /flocker-source/flocker/admin/package-files/systemd/
+      for service in $(ls *.service);
+      do
+         service_name=${service%.*}
+         systemctl stop ${service_name}
+      done
+
+      # Move systemd unit files from the clone to where systemd will look for them
+      # This uses /bin/cp instead of cp because sometimes cp is aliased to cp -i
+      # which requires confirmation
+      # This overwrites existing files (-f)
+      /bin/cp -f /flocker-source/flocker/admin/package-files/systemd/* /etc/systemd/system/multi-user.target.wants
+
+      # Reload systemd, so that it can find new or changed units:
+      systemctl daemon-reload
+
+      # Start systemd units
+      cd /flocker-source/flocker/admin/package-files/systemd/
+      for service in $(ls *.service);
+      do
+         service_name=${service%.*}
+         if [ "$(systemctl is-enabled ${service_name})" == 'enabled' ]
```

Comment:
What about new units? They won't be enabled, even if they should be. I'm not sure what can be done about this. Maybe just a note.